validator/src/bin/solana-test-validator.rs

```diff
@@ -213,6 +213,56 @@ fn main() {
         }
     }
 
+    let mut upgradeable_programs_to_load = vec![];
+    if let Some(values) = matches.values_of("upgradeable_program") {
+        let values: Vec<&str> = values.collect::<Vec<_>>();
+        for address_program_upgrade_authority in values.chunks(3) {
```

Comment:
I think this loop can be rewritten with itertools like
```rust
for (address, program, upgrade_authority) in values.into_iter().tuples::<(&str, &str, &str)>() {
```

maybe will eliminate additional vector creation

Yet not sure if this improves readability of the code

Reply:
Thanks, it does seem a bit better. However, we already use this pattern several other places, so I'm inclined to do any such refactoring in a separate PR.

Reply:
yeah, looks reasonable. I can do it if you want